app/upgrades/v10/upgrades.go

```diff
@@ -43,13 +46,9 @@ func CreateUpgradeHandler(
 		// 	fromVM[moduleName] = eachModule.ConsensusVersion()
 		// }
 
-		// TODO: should have a way to read from current node home
-		userHomeDir, err := os.UserHomeDir()
-		if err != nil {
-			stdlog.Println("Failed to get home dir %2", err)
-		}
-		nodeHome := userHomeDir + "/.sovereign/config/genesis.json"
-		appState, _, err := genutiltypes.GenesisStateFromGenFile(nodeHome)
+		nodeHome := cast.ToString(appOpts.Get(flags.FlagHome))
```

Comment:
@jstr1121 does this require the node operator to configure their node in a specific way?